load_easybuild_module.sh

```diff
@@ -0,0 +1,108 @@
+# Script to load the environment module for a specific version of EasyBuild.
+# If that module is not available yet, the current latest EasyBuild version of EasyBuild will be installed,
+# and used to install the specific EasyBuild version being specified.
+#
+# This script must be sourced, since it makes changes in the current environment, like loading an EasyBuild module.
+
+set -o pipefail
+
+if [ $# -ne 1 ]; then
+    echo "Usage: $0 <EasyBuild version>" >&2
+    exit 1
+fi
+
+# don't use $EB_VERSION, since that enables always running 'eb --version'
+EB_VERSION=${1}
+
+# make sure that environment variables that we expect to be set are indeed set
+if [ -z "${TMPDIR}" ]; then 
+    echo "\$TMPDIR is not set" >&2
+    exit 2
+fi
+
+if [ -z "${EB}" ]; then 
+    echo "\$EB is not set" >&2
+    exit 2
+fi
+
+# make sure that utility functions are defined (cfr. scripts/utils.sh script in EESSI/software-layer repo)
+type check_exit_code
+if [ $? -ne 0 ]; then
+    echo "check_exit_code function is not defined" >&2
+    exit 3
+fi
+
+echo ">> Checking for EasyBuild module..."
+
+ml_av_easybuild_out=${TMPDIR}/ml_av_easybuild.out
+module avail 2>&1 | grep -i easybuild/${EB_VERSION} &> ${ml_av_easybuild_out}
+
+if [[ $? -eq 0 ]]; then
+    echo_green ">> Module for EasyBuild v${EB_VERSION} found!"
+else
+    echo_yellow ">> No module yet for EasyBuild v${EB_VERSION}, installing it..."
+
+    EB_TMPDIR=${TMPDIR}/ebtmp
+    echo ">> Temporary installation (in ${EB_TMPDIR})..."
+    pip_install_out=${TMPDIR}/pip_install.out
+    pip3 install --prefix ${EB_TMPDIR} easybuild &> ${pip_install_out}
+
+    # keep track of original $PATH and $PYTHONPATH values, so we can restore them
+    ORIG_PATH=${PATH}
+    ORIG_PYTHONPATH=${PYTHONPATH}
+
+    echo ">> Final installation in ${EASYBUILD_INSTALLPATH}..."
+    export PATH=${EB_TMPDIR}/bin:${PATH}
+    export PYTHONPATH=$(ls -d ${EB_TMPDIR}/lib/python*/site-packages):${PYTHONPATH}
+    eb_install_out=${TMPDIR}/eb_install.out
+    ok_msg="Latest EasyBuild release installed, let's go!"
+    fail_msg="Installing latest EasyBuild release failed, that's not good... (output: ${eb_install_out})"
+    ${EB} --install-latest-eb-release 2>&1 | tee ${eb_install_out}
+    check_exit_code $? "${ok_msg}" "${fail_msg}"
+
+    eb_ec=EasyBuild-${EB_VERSION}.eb
+    ${EB} --search ${eb_ec} | grep ${eb_ec} > /dev/null
+    if [[ $? -eq 0 ]]; then
+        echo "Easyconfig found for EasyBuild v${EB_VERSION}, so installing it..."
+        ok_msg="EasyBuild v${EB_VERSION} installed, alright!"
+        fail_msg="Installing EasyBuild v${EB_VERSION}, yikes! (output: ${eb_install_out})"
+        #${EB} EasyBuild-${EB_VERSION}.eb >> ${eb_install_out} 2>&1
+        ${EB} EasyBuild-${EB_VERSION}.eb 2>&1 | tee -a ${eb_install_out}
+        check_exit_code $? "${ok_msg}" "${fail_msg}"
+    else
+        echo "No easyconfig found for EasyBuild v${EB_VERSION}"
```

Comment:
If we want to install version 4.7.2 (the currently latest release), it won't be found by the latest release, right? But because we installed it already via `--install-latest-eb-release`, it is already there.

Maybe it is sufficient to clarify that message a bit, e.g., "No easyconfig ..., it might nevertheless already installed as the latest release. Check subsequent messages regarding the module EasyBuild/${EB_VERSION}."

Reply:
It would be better to do an additional `module avail` check after `--install-latest-eb-release`, I'll look into implementing that.